Tests/BasicsTests/AuthorizationProviderTests.swift

```diff
@@ -39,20 +39,20 @@ final class AuthorizationProviderTests: XCTestCase {
             let otherPassword = UUID().uuidString
 
             // Add
-            XCTAssertNoThrow(try temp_await { callback in
+            try await safe_async { callback in
```

Comment:
Ditto, seems better to me to add `async` versions of these methods to a protocol extension of `AuthorizationProvider`.